uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -908,12 +900,12 @@ public function testManagerCanCancelOrRejectUserLeaveRequestEvenIfBalanceChangeI
     $updatedLeaveRequestData = $leaveRequestData;
     $updatedLeaveRequestData['id'] = $leaveRequest->id;
 
-    foreach (array("cancelled", "rejected") as $statusName) {
+    foreach (['cancelled', 'rejected'] as $statusName) {
       $requestNewStatus = $leaveRequestStatuses[$statusName];
       $updatedLeaveRequestData['status_id'] = $requestNewStatus;
-      $cancelledLeaveRequest = LeaveRequest::create($updatedLeaveRequestData);
+      $updatedLeaveRequest = LeaveRequest::create($updatedLeaveRequestData);
```

Comment:
Just had a second thought about this test. Not sure if its a good demonstration of the scenario. Approving and cancelling same request should be allowed normally even without the new changes because cancelling the request is a matter of releasing the balance change previously attached to the approved request thereby increasing the entitlement balance. 
I think a better scenario is still to test with two separate requests, one created approved and another created as waiting approval and trying to cancel the one in waiting approval to see if it will be possible or not.
What do you think?

Reply:
You are right. It was first cancelling the request and then rejecting the already cancelled request. However the test should test the following status transitions:

- Awaiting Approval (Balance 0) --> Cancelled
- Awaiting Approval (Balance 0) --> Rejected

Now the loop in the test does exactly this.